lib/rspec/core/runner.rb

```diff
@@ -186,6 +186,17 @@ def self.handle_interrupt
         end
       end
 
+      # @private
+      def exit_code(examples_passed=false)
+        if @world.non_example_failure
+          @configuration.error_exit_code || @configuration.failure_exit_code || 1
```

Comment:
You don't need the `|| 1`, 1 is the default configuration for `failure_exit_code`